### ISSUE TITLE
Move to functional/hooks for custom fields over class components

### ISSUE THREAD

**Issue by TellonUK** [rjsf-team/react-jsonschema-form#2120]
Are there any discussions, plans to migrate away from class components to functions/hooks in future versions?

**Comment by jacqueswho** [comment]
yes, most of the themes are using functional components and hooks. There is an effort to convert most of themes and core to typescript as possible. your contribution would be welcome
